app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java

```diff
@@ -0,0 +1,68 @@
+package org.sagebionetworks.bridge.play.controllers;
+
+import static org.sagebionetworks.bridge.Roles.DEVELOPER;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import org.sagebionetworks.bridge.models.GuidVersionHolder;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.services.SubpopulationService;
+
+import play.mvc.Result;
+
+@Controller
+public class SubpopulationController extends BaseController {
+
+    private SubpopulationService subpopService;
+    
+    @Autowired
+    public final void setSubpopulationService(SubpopulationService subpopService) {
+        this.subpopService = subpopService;
+    }
+
+    public Result getAllSubpopulations() {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        List<Subpopulation> subpopulations = subpopService.getSubpopulations(study);
+        return okResult(subpopulations);
+    }
+    public Result createSubpopulation() throws Exception {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        Subpopulation subpop = parseJson(request(), Subpopulation.class);
+        subpop = subpopService.createSubpopulation(study, subpop);
+        
+        return createdResult(new GuidVersionHolder(subpop.getGuid(), subpop.getVersion()));
+    }
+    public Result updateSubpopulation(String guid) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        Subpopulation subpop = parseJson(request(), Subpopulation.class);
+        subpop.setGuid(guid);
```

Comment:
By silently changing the guid, we could be masking a bug in the client code where they update the wrong subpop. It would be better to throw on a mismatched guid.